repository/git.go

```diff
@@ -15,22 +15,27 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
 )
 
-func cloneGitRepository(ctx context.Context, repoFullName string, localPath string, options GitHubOptions) (*git.Repository, error) {
-	url := fmt.Sprintf("https://github.com/%s.git", repoFullName)
+func cloneGitRepository(ctx context.Context, repo Repository, localPath string, options GitHubOptions) (*git.Repository, error) {
+	url := fmt.Sprintf("https://github.com/%s.git", repo.FullName())
 	logrus.WithFields(logrus.Fields{
 		"git-url":    url,
 		"local-path": localPath,
 	}).Trace("Cloning git repository")
 
+	_, token, err := githubClient(ctx, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create github client: %w", err)
+	}
+
 	gitRepo, err := git.PlainCloneContext(ctx, localPath, false, &git.CloneOptions{
 		URL: url,
 		Auth: &http.BasicAuth{
-			Username: "OctoPilot", // yes, this can be anything except an empty string
-			Password: options.Token,
+			Username: "x-access-token", // For GitHub Apps, the username must be `x-access-token`. For Personal Tokens, it doesn't matter.
+			Password: token,
 		},
 	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to clone git repository from %s to %s: %w", url, localPath, err)
+		return nil, fmt.Errorf("failed to clone git repository from %s to %s: %w, %s", url, localPath, err, token)
```

Comment:
we should avoid printing the token in the logs. do you really need this?

Reply:
Indeed, I forgot this ! Nice catch :D